substrate/frame/staking/src/ledger.rs

```diff
@@ -201,6 +201,32 @@ impl<T: Config> StakingLedger<T> {
 		}
 	}
 
+	/// Sets the ledger controller to its stash.
+	pub(crate) fn set_controller_to_stash(self) -> Result<(), Error<T>> {
+		let controller = self.controller.as_ref()
+            .defensive_proof("Ledger's controller field didn't exist. The controller should have been fetched using StakingLedger.")
+            .ok_or(Error::<T>::NotController)?;
+
+		ensure!(self.stash != *controller, Error::<T>::AlreadyPaired);
+
+		match Ledger::<T>::get(&self.stash) {
+			Some(other_ledger) => {
+				if other_ledger.stash != self.stash {
+					// stash is a controller of another ledger, fail with `Error::<T>::DoubleBonded`
+					// to avoid data inconsistencies.
+					return Err(Error::<T>::DoubleBonded);
+				}
```

Comment:
```suggestion
				// stash must not be a controller of another ledger, otherwise fail with
				// `Error::<T>::DoubleBonded` to avoid data inconsistencies.
				ensure!(other_ledger.stash == self.stash, Error::<T>::DoubleBonded);
```

Reply:
done!